design-documents/graph-ql/coverage/Cart.graphqls

```diff
@@ -0,0 +1,59 @@
+type Query {
+    cart(input: CartQueryInput): CartQueryOutput
+}
+
+input CartQueryInput {
+    cart_id: String!
+}
+
+type CartQueryOutput {
+    cart: Cart
+}
+
+type Cart {
+    id: String!
+
+    line_items_count: Int!
+    items_quantity: Float!
+
+    selected_payment_method: CheckoutPaymentMethod
+    available_payment_methods: [CheckoutPaymentMethod]!
+
+    customer: CheckoutCustomer
+    customer_notes: String
+
+    gift_cards_amount_used: Money
+    applied_gift_cards: [CartGiftCard]
+
+    is_multishipping: Boolean!
+    is_virtual: Boolean!
+}
+
+type CheckoutCustomer {
+    is_guest: Boolean!
+    email: String!
+    prefix: String
+    first_name: String!
+    last_name: String!
+    middle_name: String
+    suffix: String
+    gender: GenderEnum
+    date_of_birth: String
+    vat_number: String # Do we need it at all on storefront? Do we need more details
+}
+
+enum GenderEnum {
+    MALE
+    FEMALE
+}
+
+type CheckoutPaymentMethod {
+    code: String!
+    label: String!
+    balance: Money
+    sort_order: Int
```

Comment:
Is there a use case for exposing this value (vs having sort params in queries returning this type)?

Reply:
We thought of a use case when the client app needs to honor sort order of available payment methods during rendering.

Reply:
I think this pattern is better to be delegated to the server logic. By default `available_payment_methods` would return sorted values.

If a use case where methods should be sorted differently, it could lead to a different signature with an optional param:

```
available_payment_methods(sortDesc: Boolean)
```